build/default.conf

```diff
@@ -13,7 +13,22 @@ server {
         try_files $uri =404;
     }
 
+    # Files should be directly accessed
+    location ~* \.[a-zA-Z]+$ {
+        try_files $uri =404;
+    }
+
     location / {
+        # Demo mode params are added to URL when not yet present and DEMO is set to true
+        if ($arg_demo != "true") {
+            set $demo "${DEMO}true";
+        }
+        if ($demo = "truetrue") {
+            # This first part is required because Heroku does some internal redirects which break the rewrite otherwise
+            # This does not work in localhost as it enforces HTTPS
```

Comment:
do we have to rewrite to https here or could we just leave it at whatever was used before?

It would make more sense to have a separate block that rewrites any mode to use https (easier to understand if separated - and applying to all modes). But I don't know whether that's doable in nginx, e.g. multiple rewrite blocks applying.

Reply:
Normally it definitely works the way you are asking for the problem is that it does not work with Heroku. Somehow in there a internal redirect is happening before this one which leads to a wrong URL in this rewrite (in particular loosing the `https` and adding a wrong port). This workaround is not needed for our own servers. There we could just use:

```
rewrite ^ $uri?demo=true&session=mock permanent;
```